src/blenderbim/blenderbim/core/bsdd.py

```diff
@@ -0,0 +1,40 @@
+def load_bsdd(props, client, bsdd):
+    props.domains.clear()
```

Comment:
A minor note that things like this shouldn't be in core. E.g. `props.domains.clear()` references a concrete implementation of the Blender API. Same with stuff like `props.domains.add()` below. The core should be completely pure Python with no concretions.

Reply:
ah ok, so this should happen in tools/bsdd.py?
is there anything that shouldn't be passed to the tools Ccasses or to core and should instead happen in the operator function?

Reply:
is something like this allowed in core:
```python
props = context.scene.BIMBSDDProperties
```
or should the context be passed to the tool file?

Reply:
Core should only call tool functions, nothing else. It should read like English and be a short poem of code that describes what is going on at a high level, something like:

```python
dictionaries = bsdd.get_dictionaries()
bsdd.import_dictionaries(dictionaries)
```

or

```python
if bsdd.should_load_previews():
    dictionaries = bsdd.get_all_dictionaries()
else:
    dictionaries = bsdd.get_active_dictionaries()
bsdd.import_dictionaries(dictionaries)
```

or whatever - it's up to you, but the important point is that it's 1) not concrete 2) reads like English. This is a very simple operator, so it doesn't matter too much, but as complexity grows, having a nice high level poem to refer to can be very important.

Forcing it to be abstract ensures that all concrete interactions with Blender or IFC happens in a tool. These can be isolated, tested, and reused.